File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_094426/python/003_fibonacci/solution.py
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage:", sys.argv[0], "N")
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Usage:", sys.argv[0], "N")
        sys.exit(1)
    
    if n < 1:
        return
    
    fib_sequence = []
    a, b = 1, 1
    
    while True:
        # Add first Fibonacci number (a) if it's ≤ N
        if a <= n:
            fib_sequence.append(a)
        else:
            break
        
        # Generate next Fibonacci numbers
        c = a + b
        a = b
        b = c
        
        # Add second Fibonacci number (c) if it's ≤ N
        if c <= n:
            fib_sequence.append(c)
        else:
            break
    
    for num in fib_sequence:
        print(num)

if __name__ == "__main__":
    main()

File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_094426/typescript/003_fibonacci/solution.ts
const arg = process.argv[2];
if (!arg) {
  console.error("Usage: tsx script.ts <N>");
  process.exit(1);
}

const n = parseInt(arg, 10);

if (isNaN(n) || n < 1) {
  return;
}

let a = 1;
let b = 1;

console.log(a);
a = b;
b = a + b;

while (b <= n) {
  console.log(b);
  const next = a + b;
  a = b;
  b = next;
}